client/my-sites/stats/stats-purchase/index.tsx

```diff
@@ -173,6 +178,72 @@ const StatsPurchasePage = ( {
 						) }
 					</>
 				) }
+				{ ! isLoading && isTypeDetectionEnabled && (
+					<>
+						{
+							// a plan is owned - show a notice page for a commercial plan
+							// or a banner for a free and PWYW plans with a wizard below the notice
+							! showPurchasPage && (
+								<StatsPurchaseNoticePage
+									siteSlug={ siteSlug }
+									isCommercialOwned={ isCommercialOwned }
+									isFreeOwned={ isFreeOwned }
+									isPWYWOwned={ isPWYWOwned }
+								/>
+							)
+						}
+						{
+							// blog doesn't have any plan but is not categorised as either personal or commectial - show old purchase wizard
+							( isFreeOwned || isPWYWOwned || ( showPurchasPage && ! showSinglePurchasPage ) ) && (
```

Comment:
```suggestion
							( isFreeOwned || isPWYWOwned || options.isCommercial === null ) && (
```

This condition `( showPurchasPage && ! showSinglePurchasPage )` could be simplified to `options.isCommercial === null`, which means the site has not been classified.

Reply:
The wizard shows forever if I have a free or PWYW plan, no matter what value of `options.isCommercial` is, even when I already have bought a commercial plan.

<img width="1290" alt="buy_commercial_but_is_not_determined" src="https://github.com/Automattic/wp-calypso/assets/6869813/3d7f009a-01ed-4cd9-958f-e2660ae0b0a9">